framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -158,21 +153,102 @@ private static Pattern anchored(String regex) {
     }
 
     private static boolean isMemberSelect(String s, FlowExpressionContext context) {
-        return parseMemberSelect(s) != null;
+        return parseMemberSelect(s, context) != null;
+    }
+
+    /**
+     * Matches field accesses whose receiver is a method call expression. First of returned pair is
+     * a method call expression and the second is a field or method call expression.
+     *
+     * @param sWithFormalParamNames expression string with replaced occurences of one-based
```

Comment:
Choose argument names based on semantics, not on types.
When an argument is a string, `expr` is a better variable name than `s` (for "string").